src/hypothesis/strategies.py

```diff
@@ -1588,6 +1576,90 @@ def label(self):
     accept.__module__ = f.__module__
     return accept
 
+@composite
+def complex_numbers(
+        draw,
+        min_magnitude=0,
+        max_magnitude=None,
+        allow_infinity=None,
+        allow_nan=None
+):
+    """Returns a strategy that generates complex numbers.
+
+    This strategy draws complex numbers with constrained magnitudes.
+    The ``min_magnitude`` and ``max_magnitude`` parameters should be
+    non-negative finite or infinite floating-point-numbers, values
+    of None correspond to zero and infinite values respectively.
+
+    If min_magnitude is positive or max_magnitude is finite, it is an
+    error to enable allow_nan.
+
+    If max_magnitude is finite, it is an error to enable allow_infinity.
+
+    Examples from this strategy shrink by shrinking their magnitude as for
+    `~hypothesis.strategies.floats`, and simplifying the ratio between the
+    real and imaginary parts.
+    """
+
+    min_magnitude = try_convert(float, min_magnitude, 'min_magnitude')
+    max_magnitude = try_convert(float, max_magnitude, 'max_magnitude')
+
+    check_valid_magnitude(min_magnitude, 'min_magnitude')
+    check_valid_magnitude(max_magnitude, 'max_magnitude')
+    check_valid_interval(
+        min_magnitude, max_magnitude,
+        'min_magnitude', 'max_magnitude'
+    )
+
+    if max_magnitude == float(u'inf'):
+        max_magnitude = None
+    if min_magnitude == 0:
+        min_magnitude = None
+
+    if allow_infinity is None:
+        allow_infinity = bool(max_magnitude is None)
+    elif allow_infinity and max_magnitude is not None:
+        raise InvalidArgument(
+            'Cannot have allow_infinity=%r with max_magnitude=%r' %
+            (allow_infinity, max_magnitude)
+        )
+
+    if allow_nan is None:
+        allow_nan = bool(min_magnitude is None and max_magnitude is None)
+    elif allow_nan and (min_magnitude is not None or max_magnitude is not None):
+        raise InvalidArgument(
+            'Cannot have allow_nan=%r, min_magnitude=%r max_magnitude=%r' % (
+                allow_nan,
+                min_magnitude,
+                max_magnitude
+            )
+        )
+
+    if max_magnitude is None:
+        max_magnitude = float(u'inf')
+    if min_magnitude is None:
+        min_magnitude = 0
+
+    if allow_nan:
+        zi = draw(floats(allow_nan=True))
+        zr = draw(floats(allow_nan=True))
+        return complex(zr, zi)
+
+    zi = draw(floats(0, max_magnitude))
+    rmax = cathetus(max_magnitude, zi)
+    if zi < min_magnitude:
+        rmin = cathetus(min_magnitude, zi)
+    else:
+        rmin = 0
+    zr = draw(floats(rmin, rmax))
+
+    if draw(booleans()):
+        zr = -zr
+	if draw(booleans()):
+            return complex(zi, zr)
```

Comment:
![Ooops](https://www.emacswiki.org/pics/static/TabsSpacesBoth.png)

Reply:
Small PSA: As a general rule I'd like us to avoid jokes about violence as a response (especially to programming related opinions). 

Hmm. The code of conduct should probably mention this rule as it's come up a few times on IRC.

That being said, mixing tabs and spaces is indeed 😭